src/common/components/addressForm/addressForm.react.tsx

```diff
@@ -0,0 +1,305 @@
+import React, { useEffect, useState } from 'react';
+import angular from 'angular';
+import { react2angular } from 'react2angular';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import find from 'lodash/find';
+
+import CountrySelect from './countrySelect';
+import RegionSelect from './regionSelect';
+import TextInput from '../form/textInput';
+import FormikAutoSave from '../form/formikAutoSave';
+
+interface AddressFormProps {
+  address: Address,
+  addressDisabled?: boolean,
+  onAddressChanged: (updatedAddress: Address) => void,
+  geographiesService: any,
+  $log: any
+}
+
+export interface Address {
+  country: string,
+  locality: string,
+  region: string,
+  postalCode: string,
+  streetAddress: string,
+  extendedAddress?: string,
+  intAddressLine3?: string,
+  intAddressLine4?: string
+}
+
+interface GeographiesLink {
+  href: string,
+  rel: string,
+  type: string,
+  uri: string,
+}
+
+interface GeographiesItem {
+  "display-name": string,
+  links: GeographiesLink[],
+  name: string,
+}
+
+const componentName = 'reactAddressForm';
+
+const AddressForm = ({
+  address,
+  addressDisabled = false,
+  onAddressChanged,
+  geographiesService,
+  $log
+}: AddressFormProps) => {
+
+  const [countryName, setCountryName] = useState<string | undefined>(address.country);
+  const [countries, setCountries] = useState<GeographiesItem[]>([]);
+  const [regions, setRegions] = useState<GeographiesItem[]>([]);
+
+  const [loadingCountriesError, setLoadingCountriesError] = useState<boolean>(false);
+  const [loadingRegionsError, setLoadingRegionsError] = useState<boolean>(false);
+
+  useEffect(() => {
+    loadCountries();
+  }, []);
+
+  const dropdownSortComparator = (a: GeographiesItem, b: GeographiesItem) => {
+    if(a['display-name'] < b['display-name']) return -1;
+    if(a['display-name'] > b['display-name']) return 1;
+    return 0;
+  };
+
+  const AddressSchema = Yup.object().shape({
+    country: Yup.string()
+      .required('You must select a country'),
+    streetAddress: Yup.string()
+      .max(200, 'This field cannot be longer than 200 characters')
```

Comment:
Can this use the `maxLength` property instead of hard-coded `200`?

Reply:
I don't think the `maxLength` property on the input itself is accessible in this scope.

What I've also noticed is that setting `maxLength` on the input field prevents the user from entering any more characters than that limit.  So I don't think this error will even show up at all.  I think the input on its own does a good job of preventing the max length from being exceeded.